iidm/iidm-impl/src/main/java/com/powsybl/iidm/network/impl/extensions/BranchObservabilityImpl.java

```diff
@@ -0,0 +1,120 @@
+/**
+ * Copyright (c) 2021, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.iidm.network.impl.extensions;
+
+import com.powsybl.commons.extensions.AbstractExtension;
+import com.powsybl.iidm.network.Branch;
+import com.powsybl.iidm.network.extensions.BranchObservability;
+import com.powsybl.iidm.network.extensions.ObservabilityQuality;
+
+/**
+ * @author Thomas Adam <tadam at silicom.fr>
+ */
+public class BranchObservabilityImpl<B extends Branch<B>> extends AbstractExtension<B>
+        implements BranchObservability<B> {
+
+    private boolean observable;
+
+    private ObservabilityQuality<B> qualityP1;
+
+    private ObservabilityQuality<B> qualityP2;
+
+    private ObservabilityQuality<B> qualityQ1;
+
+    private ObservabilityQuality<B> qualityQ2;
+
+    public BranchObservabilityImpl(B component, boolean observable) {
+        super(component);
+        this.observable = observable;
+    }
+
+    public BranchObservabilityImpl(B component, boolean observable,
+                                   double standardDeviationP1, Boolean redundantP1,
+                                   double standardDeviationP2, Boolean redundantP2,
+                                   double standardDeviationQ1, Boolean redundantQ1,
+                                   double standardDeviationQ2, Boolean redundantQ2) {
+        this(component, observable);
+        this.qualityP1 = new ObservabilityQualityImpl<>(standardDeviationP1, redundantP1);
+        this.qualityP2 = new ObservabilityQualityImpl<>(standardDeviationP2, redundantP2);
+        this.qualityQ1 = new ObservabilityQualityImpl<>(standardDeviationQ1, redundantQ1);
+        this.qualityQ2 = new ObservabilityQualityImpl<>(standardDeviationQ2, redundantQ2);
+    }
+
+    public boolean isObservable() {
+        return observable;
+    }
+
+    @Override
+    public BranchObservability<B> setObservable(boolean observable) {
+        this.observable = observable;
+        return this;
+    }
+
+    @Override
+    public ObservabilityQuality<B> getQualityP1() {
+        return qualityP1;
+    }
+
+    @Override
+    public BranchObservability<B> setQualityP1(double standardDeviation, Boolean redundant) {
```

Comment:
Like in InjectionObservability, I suppose we're trying to avoid creating new objects if quality changes everywhere? Then we could also add a `setQualityP1(double)`, to avoid boxing and creating Boolean on each branch at each quality change, even if the redundant value (often?) remains the same. Well I must admit that boxing boolean is not very significant!

Reply:
To reinforce what @flo-dup said, I think we should have two methods:
one `setQualityP1(double standardDeviation, boolean redundant)` and one `setQualityP1(double standardDeviation)`. Only the field in the class should be `Boolean` (same for all optional `redundant` fields).